### DISCUSSION TITLE
Source generator

### DISCUSSION THREAD

**Discussion by bastiennoel93** [christianhelle/refitter#424] [Q&A]
Hello,

When I open my visual studio, files are generated. 
But when I delete them, after opening, even if I restore or rebuild, the files are not generated. 
I have to close and open my visual studio for them to regenerate. 

Have you had this problem?

**Comment by christianhelle** [comment]
@bastiennoel93 Thanks for taking the time to report this. 

There was a known issue in a previous release of Refitter where the Source Generator stopped working due to a bad dependency update. The source generator in lastest stable version, [v1.1.3](https://www.nuget.org/packages/Refitter.SourceGenerator/1.1.3) works fine after I just tested it right now

I created an empty .NET console app and added the Refit and Refitter.SourceGenerator package references. The csproj looks like this

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Refit" Version="7.1.2" />
    <PackageReference Include="Refitter.SourceGenerator" Version="1.1.3">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

I added a .refitter file that points to the Swagger Petstore v3 example. It looks like this

```json
{
  "openApiPath": "https://petstore3.swagger.io/api/v3/openapi.json",
  "namespace": "Petstore"
}
```

When I build the project, I get the following, it generates a folder called `Generated` with a single file in it called `petstore.g.cs`

My working folder looks like this:

```sh
total 24
drwxr-xr-x  3 christianhelle  staff   96 Jul 23 16:37 Generated
-rw-r--r--  1 christianhelle  staff  105 Jul 23 16:28 Program.cs
-rw-r--r--  1 christianhelle  staff   99 Jul 23 16:32 petstore.refitter
-rw-r--r--  1 christianhelle  staff  582 Jul 23 16:34 refitter.sourcegen.sample.csproj
```

**Comment by christianhelle** [comment]
The issue appears in [v1.1.2](https://www.nuget.org/packages/Refitter.SourceGenerator/1.1.2) and I updated the package in nuget.org to display it as deprecated

<img width="871" alt="image" src="https://github.com/user-attachments/assets/400a9a5a-488a-4be1-9caf-60be68a77512">

When I use v1.1.2 in a project like this

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Refit" Version="7.1.2" />
    <PackageReference Include="Refitter.SourceGenerator" Version="1.1.2">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

The build output is the following

```
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  Restored /Users/christianhelle/projects/refitter.sourcegen.sample/refitter.sourcegen.sample.csproj (in 104 ms).
CSC : warning CS9057: The analyzer assembly '/Users/christianhelle/.nuget/packages/refitter.sourcegenerator/1.1.2/analyzers/dotnet/cs/Refitter.SourceGenerator.dll' references version '4.10.0.0' of the compiler, which is newer than the currently running version '4.8.0.0'. [/Users/christianhelle/projects/refitter.sourcegen.sample/refitter.sourcegen.sample.csproj]
  refitter.sourcegen.sample -> /Users/christianhelle/projects/refitter.sourcegen.sample/bin/Debug/net8.0/refitter.sourcegen.sample.dll

Build succeeded.

CSC : warning CS9057: The analyzer assembly '/Users/christianhelle/.nuget/packages/refitter.sourcegenerator/1.1.2/analyzers/dotnet/cs/Refitter.SourceGenerator.dll' references version '4.10.0.0' of the compiler, which is newer than the currently running version '4.8.0.0'. [/Users/christianhelle/projects/refitter.sourcegen.sample/refitter.sourcegen.sample.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.57
```

and no output file is created

**Comment by christianhelle** [comment]
> I have to close and open my visual studio for them to regenerate.

@bastiennoel93  Sorry! I missed this detail in my previous responses. No, I don't think I have seen this before. One thing you could do to force re-generation is to shut down the [build-server](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build-server) by calling `dotnet build-server shutdown` from the command line in the projects working folder

I spend most of my time in Rider which only triggers the source generator upon build. Visual Studio triggers the source generator by making changes to the .refitter file. I guess triggering the source generator based on file changes are better than something triggered by the build, but I just got use to it
